src/qml_generator/generator.cpp

```diff
@@ -0,0 +1,67 @@
+#include "inja.hpp"
+#include <iostream>
+
+
+void createMain(inja::json& data)
+{
+    inja::Environment env;
+    inja::Template templ = env.parse_template("template_main_qml.txt");
+    env.write(templ, data, "main.qml");
+    inja::render_to(std::cout, templ.content, data);
+}
+
+void createTerminal(inja::json& data)
+{
+    inja::Environment env;
+    inja::Template templ = env.parse_template("template_terminal_qml.txt");
+    env.write(templ, data, "Terminal.qml");
+    inja::render_to(std::cout, templ.content, data);
+}
+
+void createRequestPermission(inja::json& data)
+{
+    inja::Environment env;
+    inja::Template templ = env.parse_template("template_request_permission_qml.txt");
+    env.write(templ, data, "RequestPermission.qml");
+    inja::render_to(std::cout, templ.content, data);
+}
+
+int main()
+{
+    auto filldataQt5 = [](inja::json & data)
+    {
+        data["import"]["ContourTerminal"] = "Contour.Terminal 1.0 \n";
+        data["import"]["QtQuick"] = "QtQuick 2.6 \n";
+        data["import"]["QtQuickControls"] = "QtQuick.Controls 2.6 \n";
+        data["import"]["QtQuickLayouts"] = "QtQuick.Layouts 1.11 \n";
+        data["import"]["QtQuickWindow"] = "QtQuick.Window 2.6 \n";
+        data["import"]["QtQuicklabs"] = "Qt.labs.platform 1.1 \n";
+        data["import"]["QtMultimedia"] = "QtMultimedia 5.15 \n";
+        data["import"]["QtGraphicalEffects"] = "Qt5Compat.GraphicalEffects 6.0 \n";
+
+    };
+
+    auto filldataQt6 = [](inja::json & data)
+    {
+        data["import"]["ContourTerminal"] = "Contour.Terminal  \n";
+        data["import"]["QtQuick"] = "QtQuick \n";
+        data["import"]["QtQuickControls"] = "QtQuick.Controls \n";
+        data["import"]["QtQuickLayouts"] = "QtQuick.Layouts \n";
+        data["import"]["QtQuickWindow"] = "QtQuick.Window \n";
+        data["import"]["QtQuicklabs"] = "Qt.labs.platform \n";
+        data["import"]["QtMultimedia"] = "QtMultimedia  \n";
+        data["import"]["QtGraphicalEffects"] = "QtGraphicalEffects \n";
+
+    };
+    inja::json data;
+    #ifdef GENERATE_QT5
+    filldataQt5(data);
+    #else
+    filldataQt6(data);
```

Comment:
sounds like what you want (to stay in line with the other helper methods), is something like `createModuleNames(data)`, a single global function, and then have the `#ifdef GENERATE_QT5` inside of this body.

All of the above functions are private, and can thus be moved inside an anonymous namespace to hide their symbols and mark them as TU-private (`namespace { ... }`)